cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueClientFactory.java

```diff
@@ -84,6 +86,16 @@ public HBaseQueueClientFactory(CConfiguration cConf, Configuration hConf, HBaseT
   }
 
   @Override
+  public void initContext(Id.Run run) {
+    queueAdmin.initContext(run);
```

Comment:
It is okay to set the run on singleton QueueAdmin since there will be only one run associated with a QueueAdmin, right?